src/Options/Generic.hs

```diff
@@ -845,6 +847,23 @@ getRecord desc = liftIO (Options.customExecParser defaultParserPrefs info)
     header = Options.header (Data.Text.unpack desc)
     info = Options.info parseRecord header
 
+-- | Marshal any value that implements `ParseRecord` from the commmand line
+-- alongside an io action that prints the help message.
+getWithHelp
+    :: (MonadIO io, ParseRecord a)
+    => Text
+    -- ^ Program description
+    -> io (a, io ())
+    -- ^ (options, io action to print help message)
+getWithHelp desc =
+  (,)
```

Comment:
Minor suggestion: perhaps use `do` notation here because it's shorter and possibly easier to read for people less familiar with `Applicative`:

```haskell
getWithHelp desc = do
    a <- liftIO (Options.customExecParser defaultParserPrefs info)
    return (a, help)
```